lib/kramdown/converter/html.rb

```diff
@@ -147,6 +147,14 @@ def convert_header(el, indent)
         if @options[:auto_ids] && !attr['id']
           attr['id'] = generate_id(el.options[:raw_text])
         end
+
+        if @options[:auto_ids] && @options[:header_links]
```

Comment:
Option 'audo_ids' is not necessary for this feature to work, the header just needs to have an 'id' attribute set.